File: C#/5/5_1.cs
void AnalyzeData(List<int> data)
{
    if (data == null || data.Count == 0)
    {
        Console.WriteLine("Data is empty or undefined.");
        return;
    }

    int total = 0;
    foreach (int value in data)
    {
        if (value <= 0)
        {
            Console.WriteLine($"Non-positive value encountered: {value}");
            continue;
        }

        total += value % 2 == 0 ? value : -value;
    }
    
    Console.WriteLine($"Final total: {total}");
}

File: JavaScript/5/5_1.js
function isValidData(data) {
    return data && data.length > 0;
  }
 
  function processValue(value, total) {
    if (value > 0) {
      if (value % 2 === 0) {
        return total + value;  
      } else {
        return total - value;  
      }
    } else {
      console.log("Non-positive value encountered: " + value);
      return total;
    }
  }
 
  function analyzeData(data) {
    if (!isValidData(data)) {
      console.error("Data is empty or undefined.");
      return;
    }
 
    let total = 0;
    for (let i = 0; i < data.length; i++) {
      total = processValue(data[i], total);  
    }
 
    console.log("Final total: " + total);
  }